src/auth/strategies/at.strategy.ts

```diff
@@ -1,23 +1,63 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, UnauthorizedException } from '@nestjs/common';
 import { PassportStrategy } from '@nestjs/passport';
+import { Request } from 'express';
 import { ExtractJwt, Strategy } from 'passport-jwt';
+import { decode, verify } from 'jsonwebtoken';
 import { JwtPayload } from 'src/@types/auth';
 import { User } from 'src/@types/users';
 import { UserService } from '../../user/user.service';
+import { AuthService } from '../auth.service';
+import { Token } from '../../@types/auth';
 
 @Injectable()
 export class JwtAtStrategy extends PassportStrategy(Strategy, 'jwt') {
-  constructor(private userService: UserService) {
+  constructor(
+    private userService: UserService,
+    private authService: AuthService,
+  ) {
     super({
       jwtFromRequest: ExtractJwt.fromAuthHeaderAsBearerToken(),
       secretOrKey: process.env.JWT_SECRET,
-      ignoreExpiration: false,
+      ignoreExpiration: true,
+      passReqToCallback: true,
     });
   }
 
-  async validate({ id }: JwtPayload): Promise<User> {
+  async validate(req: Request, payload: JwtPayload): Promise<User> {
+    const { id } = payload;
+    const at = req.headers.authorization.replace('Bearer ', '');
```

Comment:
First of all, it's not time-related variable but it's named `at`.
And how about creating an utility that splits string and picks a key?